app/src/controllers/client/shop/services/ShopService.ts

```diff
@@ -62,6 +64,16 @@ const ShopService: ShopServiceInterface = {
     return { shops, totalCount }
   },
 
+  async fetchShopsByNameWithTotalCount(user, name, page = 1, order = OrderBy.DESC) {
+    const shops = await ShopRepository.fetchShopsByName(name, page, order)
+    if (!shops) {
+      Logger.debug('Shop does not exist')
+      throw new NotFoundError()
+    }
```

Comment:
no need to throw an error if because this should always return an array, if there are no matches then we should just return an empty array